src/ExpressionEvaluator/CSharp/Test/ResultProvider/FullNameTests.cs

```diff
@@ -321,5 +327,483 @@ public class @class : @interface<@class>
+    ret
+  } // end of method Type::.ctor
+
+} // end of class Type
+
+.class public auto ansi beforefieldinit '<>Mangled'
+       extends [mscorlib]System.Object
+{
+  .field public bool y
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor(class Type s) cil managed
+  {
+    ldarg.0
+    ldc.i4.1
+    stfld      bool '<>Mangled'::y
+    ldarg.0
+    call       instance void [mscorlib]System.Object::.ctor()
+    ret
+  } // end of method '<>Mangled'::.ctor
+
+} // end of class '<>Mangled'
+";
+
+            ImmutableArray<byte> assemblyBytes;
+            ImmutableArray<byte> pdbBytes;
+            CSharpTestBase.EmitILToArray(il, appendDefaultHeader: true, includePdb: false, assemblyBytes: out assemblyBytes, pdbBytes: out pdbBytes);
+            var assembly = ReflectionUtilities.Load(assemblyBytes);
+
+            var value = CreateDkmClrValue(assembly.GetType("Type").Instantiate());
+
+            var root = FormatResult("o", value);
+            var children = GetChildren(root);
+            Verify(children,
+                EvalResult("y", "true", "bool", null, DkmEvaluationResultFlags.Boolean | DkmEvaluationResultFlags.BooleanTrue),
+                EvalResult("Raw View", null, "", "o, raw", DkmEvaluationResultFlags.Expandable | DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Data));
+
+            var grandChildren = GetChildren(children.Last());
+            Verify(grandChildren,
+                EvalResult("x", "false", "bool", "o.x", DkmEvaluationResultFlags.Boolean));
+        }
+
+        [Fact]
+        public void BackTick_NonGenericType()
+        {
+            var il = @"
+.class public auto ansi beforefieldinit 'C`1' extends [mscorlib]System.Object
+{
+  .field public static int32 'x'
+
+  .method public hidebysig specialname rtspecialname instance void  .ctor() cil managed
+  {
+    ldarg.0
+    call       instance void [mscorlib]System.Object::.ctor()
+    ret
+  }
+}
+";
+
+            ImmutableArray<byte> assemblyBytes;
+            ImmutableArray<byte> pdbBytes;
+            CSharpTestBase.EmitILToArray(il, appendDefaultHeader: true, includePdb: false, assemblyBytes: out assemblyBytes, pdbBytes: out pdbBytes);
+            var assembly = ReflectionUtilities.Load(assemblyBytes);
+
+            var value = CreateDkmClrValue(assembly.GetType("C`1").Instantiate());
+
+            var root = FormatResult("o", value);
+            var children = GetChildren(root);
+            Verify(children,
+                EvalResult("Static members", null, "", null, DkmEvaluationResultFlags.Expandable | DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Class));
+            Verify(GetChildren(children.Single()),
+                EvalResult("x", "0", "int", null));
+        }
+
+        [Fact]
+        public void BackTick_GenericType()
+        {
+            var il = @"
+.class public auto ansi beforefieldinit 'C`1'<T> extends [mscorlib]System.Object
```

Comment:
Yes this is the normal case.  It is here for contrast with the following test and to emphasize coverage.